tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -0,0 +1,273 @@
+// <copyright file="LiveDebugger.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent.DiscoveryService;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Debugger.Configurations;
+using Datadog.Trace.Debugger.Configurations.Models;
+using Datadog.Trace.Debugger.Helpers;
+using Datadog.Trace.Debugger.Models;
+using Datadog.Trace.Debugger.PInvoke;
+using Datadog.Trace.Debugger.ProbeStatuses;
+using Datadog.Trace.Debugger.Sink;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Debugger
+{
+    internal class LiveDebugger
+    {
+        private static readonly Lazy<LiveDebugger> LazyInstance = new Lazy<LiveDebugger>(Create, isThreadSafe: true);
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(LiveDebugger));
+
+        private readonly ImmutableDebuggerSettings _settings;
+        private readonly IDiscoveryService _discoveryService;
+        private readonly IConfigurationPoller _configurationPoller;
+        private readonly IDebuggerSink _debuggerSink;
+        private readonly ILineProbeResolver _lineProbeResolver;
+        private readonly List<ProbeDefinition> _unboundProbes;
+        private readonly IProbeStatusPoller _probeStatusPoller;
+        private readonly object _locker = new();
+
+        private LiveDebugger(ImmutableDebuggerSettings settings, IDiscoveryService discoveryService, IConfigurationPoller configurationPoller, ILineProbeResolver lineProbeResolver, IDebuggerSink debuggerSink, IProbeStatusPoller probeStatusPoller)
+        {
+            _settings = settings;
+            _discoveryService = discoveryService;
+            _configurationPoller = configurationPoller;
+            _lineProbeResolver = lineProbeResolver;
+            _debuggerSink = debuggerSink;
+            _probeStatusPoller = probeStatusPoller;
+            _unboundProbes = new List<ProbeDefinition>();
+        }
+
+        public static LiveDebugger Instance => LazyInstance.Value;
+
+        private static LiveDebugger Create()
+        {
+            var source = GlobalSettings.CreateDefaultConfigurationSource();
+            var settings = ImmutableDebuggerSettings.Create(DebuggerSettings.FromSource(source));
+            if (!settings.Enabled)
+            {
+                return Create(settings, null, null, null, null, null);
+            }
+
+            var apiFactory = DebuggerTransportStrategy.Get(new ExporterSettings().AgentUri);
```

Comment:
There's a _lot_ of disparate `ExporterSettings` getting created throughout this PR, none of which necessarily correspond to the _actual_ `ExporterSettings` used by the tracer (because the tracer has to support config in code). They also all seem to assume tcp/ip, which is not always the case.

Reply:
Yes, you're right, we need to refactor it a bit, and re-use same settings.
I've removed this particular usage at 0e23585745d965ea143036238a6033ed931c09bd, I think there are no more usage, except of `DebuggerSettings`

Reply:
This should do for now, just be aware you'll need to remove that new `AgentUri` property you added later!

Reply:
Yes, sure, will remove it in on of the removing `ImmutableDebuggerSettings`, or supporting `UDS/Named pipes` PR.